### ISSUE TITLE
Clarify disallowed operations in AIO completion callbacks

### ISSUE THREAD

**Issue by EvanBalster** [nanomsg/nng#1428]
Hello —

I've been designing an application that uses AIO for various send/recv operations, and encountered some pitfalls in my use of completion callbacks.

My comments pertain to this doc page:  https://nng.nanomsg.org/man/v0.6.0/nng_aio_alloc.html

It should be clarified that AIO callbacks should not open or close any sockets, listeners or dialers.  It might also be helpful to comment on whether it's appropriate to allocate new AIO handles in these callbacks.

**Comment by yotann** [comment]
It'd be nice to have a separate document that explains how to avoid deadlock, not just a paragraph in `nng_aio_alloc(3)`. In particular, it should explain that some operations (like `nng_aio_wait`) may block until a queued task is complete; if all task threads are currently blocked on `nng_aio_wait`, deadlock will occur.

**Comment by gdamore** [comment]
the shortcomings in the docs are evident -- we need a real guide, not just a reference manual.   for now i'm going to update the reference manual with a bit more detail.
